File: python/hash/disguise.py
# 위장(https://programmers.co.kr/learn/courses/30/parts/12077)
# clothes	                                                                        return
# [[yellow_hat, headgear], [blue_sunglasses, eyewear], [green_turban, headgear]]	5
# [[crow_mask, face], [blue_sunglasses, face], [smoky_makeup, face]]	            3

def solution(clothes):
    answer = 1
    dic = {}
    for i in clothes:
        if i[1] not in dic:
            dic[i[1]] = 2
        else:
            dic[i[1]] += 1
    cnt = list(dic.values())
    if len(cnt)==1:
        return cnt[0]-1
    for i in cnt:
        answer *= i
    answer -= 1
    return answer

File: javascript/hash/disguise.js
function solution(clothes) {
    var answer = 1;
    const dict = {}
    for (let i=0; i<clothes.length; i++) {
        let c = clothes[i][0]
        let k = clothes[i][1]
        if (k in dict) {
            dict[k] += 1
        } else {
            dict[k] = 1
        }
    }
    for (let key in dict) {
        answer *= dict[key] + 1
    }
    return answer-1;
}

console.log(solution([["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"]]))
console.log(solution([["crow_mask", "face"], ["blue_sunglasses", "face"], ["smoky_makeup", "face"]]))